Updated the multi_site_inputs_parser.py to allow scenario specific electric tariffs

JesseKaczmarski opened the pull request:
This fork of the repository includes an updated version of `src.multi_site_input_parser.py`. In particular, it allows the addition of a column in the multi-scenario input file titled `rate_file` that allows a specific electric tariff to be specified for each scenario. Below is an example of some inputs:

| description  | load_file | rate_file |
| ------------- | ------------- | ------------- |
| 87104  | 87104_load_kw.csv  | 87104_electric_tariff.json  |
| 87106  | 87106_load_kw.csv  | 87106_electric_tariff.json  |

The multi-scenario example in the original repository notes that you must post a singular electric rate tariff, but this allows any number of tariffs to be posted as long as they are in the input file under this column header.

The readme file is probably something you don't want to commit, but if you find this a useful addition to the parser file then let me know. Everything was tested and works on my end.

NLaws commented on src/multi_site_inputs_parser.py:
```diff
@@ -130,6 +130,11 @@ def multi_site_csv_parser(path_to_csv, api_url, API_KEY, n_sites=None):
         add_load_profile_inputs(site_inputs, posts[-1], path_to_load_files=path_to_load_files)
+        
+        path_to_rate_files = "./electric_rates"
+        for x in df['rate_file'].tolist():
```
@JesseKaczmarski What happens if 'rate_file' is not in the DataFrame columns? I think that this block needs to be wrapped with an `if "rate_file" in df.columns`.

NLaws commented on src/multi_site_inputs_parser.py:
```diff
@@ -130,6 +130,11 @@ def multi_site_csv_parser(path_to_csv, api_url, API_KEY, n_sites=None):
+        
+        path_to_rate_files = "./electric_rates"
+        for x in df['rate_file'].tolist():
+            rate_x = json.load(open(os.path.join(path_to_rate_files, x), "r"))
```
Since the `rate_file`s must be JSON, and we put them in the `urdb_response`, maybe we can call `rate_file` something more descriptive? Something like `urdb_json_file`.

NLaws commented on README.md:
```diff
@@ -1,5 +1,16 @@
 # REopt API Analysis using Python
 
+## Updated for Multi-Scenario and Multi-Tariff Inputs
+This fork of the repository includes an updated version of `src.multi_site_input_parser.py`. In particular, it allows the addition of a column in the multi-scenario input file titled `rate_file` that allows a specific electric tariff to be specified for each scenario. Below is an example of some inputs:
```
Thank you for updating the README! Can you please modify the text to reflect what you would want as a new user of the capability (and remove the mention of the fork)?

NLaws reviewed (changes requested):
Please see inline review comments.

JesseKaczmarski commented:
Hey there @NLaws,

Thanks for such a quick reply! Glad you brought up the logical tests, otherwise the program would not run without a column titled `rate_file`.  I have updated code to include logical tests, and to even allow for using either the specified json file or the urdb label if one is not specified for that specific scenario. I also reworked the readme as requested. Let me know what you think!

JesseKaczmarski commented:
Clarification on the language in my commit: the elif statement breaks the program if both a urdb_json_file and urdb_label column are present, _but both are empty_.

NLaws commented on src/multi_site_inputs_parser.py:
```diff
@@ -130,6 +130,19 @@ def multi_site_csv_parser(path_to_csv, api_url, API_KEY, n_sites=None):
+        path_to_rate_files = "./electric_rates"
+        if "urdb_json_file" in df.columns:
+            if pd.isnull(df.loc[i,"urdb_json_file"]) == False:
+                rate_i = json.load(open(os.path.join(path_to_urdb_json_files, df["urdb_json_file"][i]), "r"))
```
should `path_to_urdb_json_files` be `path_to_rate_files`? (Or vice-versa?)

NLaws commented on src/multi_site_inputs_parser.py:
```diff
@@ -130,6 +130,19 @@ def multi_site_csv_parser(path_to_csv, api_url, API_KEY, n_sites=None):
+            if pd.isnull(df.loc[i,"urdb_json_file"]) == False:
+                rate_i = json.load(open(os.path.join(path_to_rate_files, df["urdb_json_file"][i]), "r"))
+                posts[i]["Scenario"]["Site"]["ElectricTariff"]["urdb_response"] = rate_i
+            elif pd.isnull(df.loc[i,"urdb_json_file"]) == True and pd.isnull(df.loc[i,"urdb_label"]) == True:
```
What if a scenario has the ElectricTariff specified via inputs other than the `urdb_response` or `urdb_label`? (A user can provide [any of these sets](https://github.com/NREL/REopt_API/blob/fbc70f3b0cdeec9ee220266d6b3b0c5d64f257a6/reo/nested_inputs.py#L72)). I think that we can remove this `elif` block (otherwise we must account for every combination possible for specifying the `ElectricTariff`, which can change and thus will require maintenance). Is that okay with you @JesseKaczmarski ?

JesseKaczmarski replied:
Yes, if we remove the elif statement it will just revert to how the program operated before. I'll make the change!

NLaws replied:
Thank you @JesseKaczmarski ! I will approve and merge.
